### PR TITLE
Add support for RegExp operator

### PR DESCRIPTION
**Pull request by superkhau**
Connect to strongloop/loopback-datasource-juggler#552

### CODE REVIEW

File: lib/mysql.js [hunk 1]
```diff
@@ -459,6 +459,22 @@ MySQL.prototype.ping = function(cb) {
   this.execute('SELECT 1 AS result', cb);
 };
 
+MySQL.prototype.buildExpression = function(columnName, operator, columnValue, propertyValue) {
+  if (operator === 'regexp' && typeof columnValue === 'string') {
```
Comment (raymondfeng) [inline lib/mysql.js]: Assuming the normalization of the operand of `regexp` is done in juggler, we can clean up the code here.

**Comment by superkhau** [comment]
@raymondfeng Review please.

File: lib/mysql.js [hunk 2]
```diff
@@ -6,6 +6,7 @@ var mysql = require('mysql');
 var SqlConnector = require('loopback-connector').SqlConnector;
 var ParameterizedSQL = SqlConnector.ParameterizedSQL;
 var EnumFactory = require('./enumFactory').EnumFactory;
+var utils = require('loopback-datasource-juggler/lib/utils');
```
Comment (raymondfeng) [inline lib/mysql.js]: This is invalid. `loopback-datasource-juggler` is a dev dep.

File: lib/mysql.js [hunk 3]
```diff
@@ -459,6 +460,19 @@ MySQL.prototype.ping = function(cb) {
   this.execute('SELECT 1 AS result', cb);
 };
 
+MySQL.prototype.buildExpression = function(columnName, operator, columnValue,
+    propertyValue) {
+  if (operator === 'regexp') {
+    // strip regex flags because mysql doesn't allow them
+    var regex = utils.getRegExpExpression(columnValue);
```
Comment (raymondfeng) [inline lib/mysql.js]: What does `utils.getRegExpExpression(columnValue);` do? `propertyValue` should already have the value of RegExp.

**Comment by superkhau** [comment]
@raymondfeng Issues addressed.

File: lib/mysql.js [hunk 4]
```diff
@@ -459,6 +459,16 @@ MySQL.prototype.ping = function(cb) {
   this.execute('SELECT 1 AS result', cb);
 };
 
+MySQL.prototype.buildExpression = function(columnName, operator, columnValue,
+    propertyValue) {
+  if (operator === 'regexp')
+    return new ParameterizedSQL(columnName + ' REGEXP ?', [columnValue.source]);
```
Comment (raymondfeng) [inline lib/mysql.js]: Ignoring the `i` flag silently is not a good idea. We should either report the error or at least print out a warning.

**Comment by superkhau** [comment]
@raymondfeng LGTY?

File: lib/mysql.js [hunk 5]
```diff
@@ -459,6 +459,17 @@ MySQL.prototype.ping = function(cb) {
   this.execute('SELECT 1 AS result', cb);
 };
 
+MySQL.prototype.buildExpression = function(columnName, operator, operatorValue,
+    propertyDefinition) {
+  if (operator === 'regexp')
+    return new ParameterizedSQL(columnName + ' REGEXP ?',
```
Comment (raymondfeng) [inline lib/mysql.js]: I believe we agreed to throw an error if the `i` flag is set.

**Comment by superkhau** [comment]
@raymondfeng PTAL again, prints warnings this time. Also added tests to make sure warnings are being printed.

**Comment by raymondfeng** [comment]
LGTM

**Pull request merged**